core/src/lib.rs

```diff
@@ -102,9 +101,10 @@ mod tests {
 
     impl TestObj {
```

Comment:
What is this TestObj used for?

Reply:
It's a mock object and is used for applying a trait to it. In real case it has to be library-consumer owned object